site/app/models/gradeable/Gradeable.php

```diff
@@ -238,32 +244,25 @@ public function getComponent($component_id) {
 
     /**
      * Loads the autograding config file at $this->autograding_config into an array, or null if error/not found
-     * @return array|bool|null
+     * @return AutogradingConfig|null
      */
     private function loadAutogradingConfig() {
         $course_path = $this->core->getConfig()->getCoursePath();
 
         try {
             $details = FileUtils::readJsonFile(FileUtils::joinPaths($course_path, 'config', 'build',
                 "build_{$this->id}.json"));
+
+            // If the file could not be found, the result will be false, so don't
+            //  create the config if the file can't be found
+            if ($details !== false) {
+                return new AutogradingConfig($this->core, $details);
```

Comment:
At least with the original model, autograding configs were cached on a per-gradeable basis to cut down on file reading overhead (see GradeableAutogradingConfig.php:11-15). Would be nice to do similar here.

Reply:
Actually I've just realized this is not GradedGradeable... disregard this